lib/string.gd

```diff
@@ -860,6 +860,85 @@ BindGlobal("BHINT", MakeImmutable("\>\<"));
 
 DeclareGlobalFunction("StringOfMemoryAmount");
 
+#############################################################################
+##
+##  <#GAPDoc Label="StringFormatted">
+##  <ManSection>
+##  <Func Name="StringFormatted" Arg='string, data...'/>
+##  <Func Name="PrintFormatted" Arg='string, data...'/>
+##  <Func Name="PrintToFormatted" Arg='stream, string, data...'/>
+##
+##  <Description>
+##  These functions perform a string formatting operation.
+##  They accept a format string, which can contain replacement fields
+##  which are delimated by braces {}.
+##  Each replacement field contains a numeric or positional argument,
+##  describing the element of <A>data</A> to replace the braces with.
+##  <P/>
+##  There are three formatting functions, which differ only in how they
+##  output the formatted string.
+##  <Ref Func="StringFormatted"/> returns the formatted string,
+##  <Ref Func="PrintFormatted"/> prints the formatted string and
+##  <Ref Func="PrintToFormatted"/> appends the formatted string to <A>stream</A>,
+##  which can be either an output stream or a filename.
+##  <P/>
+##  The arguments after <A>string</A> form a list <A>data</A> of the values to be
+##  inserted into <A>string</A>, using the following formatting rules:
```

Comment:
Nitpick: the values are not inserted into `string` (which indeed can be immutable). Perhaps 
>  ... form a list `data` of values used to substitute the replacement fields in `string`, according to the following formatting rules